pilot/pkg/xds/sds.go

```diff
@@ -120,7 +124,15 @@ func (s *SecretGen) Generate(proxy *model.Proxy, w *model.WatchedResource, req *
 		updatedSecrets = model.ConfigsOfKind(req.ConfigsUpdated, kind.Secret)
 	}
 
-	proxyClusterSecrets, err := s.secrets.ForCluster(proxy.Metadata.ClusterID)
+	secretsFromCluster := proxy.Metadata.ClusterID
+	if !features.CentralIstiodAccess {
+		// If not working in 'central istiod' mode, ignore cluster ID meta - only serve secrets from Istiod cluster.
+		// This prevents workloads in remote clusters from getting secrets in other clusters by using unverified
+		// header.
+		secretsFromCluster = s.configCluster
+	}
+
+	proxyClusterSecrets, err := s.secrets.ForCluster(secretsFromCluster)
```

Comment:
This code (with L141) is: try fetch from proxy's cluster, fallback to config cluster.

Now with CentralIstiodAccess=true, it becomes "try fetch from config cluster, try fetch from config cluster"

so I think we just want to skip this fetch on L135 and let it try only the L141?